pkg/datapath/manager.go

```diff
@@ -95,3 +105,58 @@ func (m *Manager) GetAsyncBR(jobName string) AsyncBR {
 		return nil
 	}
 }
+
+// CanAcceptNewTask checks if a new task can be accepted based on the concurrent limit.
+// This is a lightweight check that doesn't create any resources.
+func (m *Manager) CanAcceptNewTask(resume bool) bool {
+	m.trackerLock.Lock()
+	defer m.trackerLock.Unlock()
+
+	if resume {
+		return true
+	}
+
+	// Count both active tasks and reserved slots
+	return (len(m.tracker) + len(m.reservations)) < m.cocurrentNum
+}
+
+// ReserveSlot reserves a slot in the tracker before expensive operations (like GetExposed).
+// Returns ConcurrentLimitExceed if no slots are available.
+// Must be paired with either CompleteReservation or ReleaseReservation.
+func (m *Manager) ReserveSlot(taskName string) error {
+	m.trackerLock.Lock()
+	defer m.trackerLock.Unlock()
+
+	// Check if already reserved or in tracker
+	if _, exists := m.tracker[taskName]; exists {
+		return nil // Already active
+	}
+	if m.reservations[taskName] {
```

Comment:
In which case is this happening? A task is always inserted and then removed in one function.
If it is not happening, let's remove it.